src/lib/y2storage/disk_analyzer.rb

```diff
@@ -178,10 +178,14 @@ def data_for(*disks, data)
     #   All disk devices by default.
     # @return [Array<Dasd, Disk>]
     def disks_collection(disks)
-      disks = devicegraph.disk_devices if disks.empty?
+      return full_devices_list if disks.empty?
+
       # Using BlkDevice because it is necessary to search in both, Dasd and Disk.
-      disks = disks.map { |d| d.is_a?(String) ? BlkDevice.find_by_name(devicegraph, d) : d }
-      disks.compact
+      disks.map { |d| d.is_a?(String) ? BlkDevice.find_by_name(devicegraph, d) : d }.compact
+    end
+
+    def full_devices_list
```

Comment:
NP: the name of the method does not sound too much accurate to me, and maybe it can lead to misleading. Why to not include that line in `disk_collection` method?

```
    def disks_collection(disks)
      disks ||= devicegraph.disk_devices + devicegraph.software_raids + devicegraph.bcaches
      disks = disks.map { |d| d.is_a?(String) ? BlkDevice.find_by_name(devicegraph, d) : d }
      disks.compact
    end
```

Reply:
Please, also improve `disks_collection` documentation by saying that software raids and bcaches need to be analyzed since it is possible to find a Linux system installed on them (as you stated in the commit comment).

Reply:
> Why to not include that line in disk_collection method?

Because it made more sense to me having the "default_disks_collection" in a separate method, returning it directly when disks were not given, avoiding the not necessary mapping.

Furthermore, your example will not work because `[]` != `nil`. So, I prefer to have them split instead add more conditional logic to `disks_collection`. What do you think? Is it `default_disks_collection` more accurate for you?

Thank you so much.